unitTests/ui/QuickviewTest.ts

```diff
@@ -69,9 +78,59 @@ export function QuickviewTest() {
       expect(hash).toBe(result.queryUid + '.' + result.index + '.' + StringUtils.hashCode(result.uniqueId));
     });
 
-    function buildTemplate() {
-      let template = new Template(() => '<div class="coveo-quick-view-full-height"></div>');
-      return template;
-    }
+    describe('when instantiating invalid templates', () => {
+      it('should not throw an error when opening a template that throws an error', done => {
+        const badTemplate = new Template(() => {
+          throw 'Oh, the Humanity!';
+        });
+        quickview = buildQuickview(badTemplate);
+        spyOn(quickview.logger, 'warn');
+
+        quickview
+          .open()
+          .then(() => {
+            expect(quickview.logger.warn).toHaveBeenCalled();
+            done();
+          })
+          .catch(e => {
+            fail(e);
+            done();
+          });
+      });
+
+      it('should not throw an error when opening a template that returns null', done => {
+        const badTemplate = new Template(() => null);
+        quickview = buildQuickview(badTemplate);
+        spyOn(quickview.logger, 'warn');
+
+        quickview
+          .open()
+          .then(() => {
+            expect(quickview.logger.warn).toHaveBeenCalled();
+            done();
+          })
+          .catch(e => {
+            fail(e);
+            done();
+          });
+      });
+
+      it('should not throw an error when opening a template that returns undefined', done => {
+        const badTemplate = new Template(() => undefined);
+        quickview = buildQuickview(badTemplate);
+        spyOn(quickview.logger, 'warn');
+
+        quickview
```

Comment:
Would it be better to cut the repetition by creating a function called `verifyThatOpeningQuickViewDoesNotThrow(quickview, done)` that takes care of adding the spy, and the promise chain used by the tests in this `describe` block.